src/Service/FaqService.php

```diff
@@ -73,4 +73,27 @@ public function update(Uuid $identifier, array $faq): Faq
 
         return $this->repository->save($faqObj);
     }
+
+    public function list(int $limit = 50): array
+    {
+        return $this->repository->findBy(
+            ['active' => true],
+            ['createdAt' => 'DESC'],
+            $limit
+        );
+    }
+
+    public function remove(Uuid $id): void
+    {
+        $faq = $this->repository->findOneBy(['id' => $id, 'active' => true]);
+
+        if (null === $faq) {
+            throw new FaqResourceNotFoundException();
+        }
+
+        $faq->setUpdatedAt(new DateTime());
+        $faq->setActive(false);
+
+        $this->repository->save($faq);
+    }
```

Comment:
Não temos exemplos de testes assim ainda hoje, mas seria de muita valia se você pudesse garantir o funcionamento desses métodos via testes especificos, algo como `FaqServiceTest`

O que você me diz?

Reply:
Você fala em relação a todos os métodos ou apenas ao `remove`? Porque se for em relação a todos os métodos eu acho que não seria produtivo, pois estaríamos testando basicamente a mesma coisa que os testes relacionados aos controllers já testam. Se for em relação apenas ao `remove`, eu posso concordar já que nesse caso eu poderia testar se o objeto foi realmente desativado no banco de dados, algo que o teste no controller não faz.

Reply:
sim, pro `remove()` testar se o registro ainda consta no banco (só que com o deletedAt diferente de false e com a data correta).

No caso do list, acho que vale a pena testar se o parametro `$limit` de fato limita a quantidade de elementos retornados e se a ordenação de fato está `DESC`

Como os testes de endpoint não testam essas regras especificas, valeria a pena termos testes pra pero menos estes 2 casos

Reply:
Entendi, eu não tinha me atentado para esse caso do `list()`.

Criei o arquivo `FaqServiceTest` e além desses dois casos eu ainda coloquei casos para o create e o update.